vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
-                                    break
-
-                    for cve_id in cve_ids:
-                        adv_list.append(
-                            Advisory(
-                                vulnerability_id=cve_id,
-                                summary=vuln_desc,
-                                affected_packages=nearest_patched_package(
-                                    affected_purls, unaffected_purls
-                                ),
-                                references=references,
+            # attach the GHSA with severity score
+            if identifier["type"] == "GHSA":
+                for ref in references:
+                    if ref.reference_id == identifier["value"]:
+                        ref.severities = [
+                            VulnerabilitySeverity(
+                                system=SCORING_SYSTEMS["cvssv3.1_qr"],
+                                value=get_item(advisory, "severity") or "",
                             )
-                        )
-        return adv_list
-
-    @staticmethod
-    def categorize_versions(
-        package_type: str, version_range: str, all_versions: Set[str]
-    ) -> Tuple[List[str], List[str]]:
-        version_class = version_class_by_package_type[package_type]
-        version_scheme = version_class.scheme
-        version_range = VersionSpecifier.from_scheme_version_spec_string(
-            version_scheme, version_range
+                        ]
+                        # Each Node has only one GHSA, hence exit after attaching
+                        # score to this GHSA
+                        break
+        cutoff_time = get_item(advisory, "publishedAt")
+        if cutoff_time:
+            cutoff_time = dateparser.parse(cutoff_time)
+        yield AdvisoryData(
+            aliases=list(cve_ids),
+            summary=vuln_desc,
+            references=references,
+            affected_packages=affected_packages,
+            date_published=cutoff_time,
         )
-        affected_versions = []
-        unaffected_versions = []
-        for version in all_versions:
-            if version in IGNORE_VERSIONS:
-                continue
 
-            if version_class(version) in version_range:
-                affected_versions.append(version)
-            else:
-                unaffected_versions.append(version)
-        return (affected_versions, unaffected_versions)
+
+class GitHubBasicImprover(Improver):
+    @property
+    def interesting_advisories(self) -> QuerySet:
+        return Advisory.objects.filter(created_by=GitHubAPIImporter.qualified_name)
+
+    def get_inferences(self, advisory_data: AdvisoryData) -> Iterable[Inference]:
+        """
+        Generate and return Inferences for the given advisory data
+        """
+        try:
+            purl, affected_version_ranges, _ = AffectedPackage.merge(
+                advisory_data.affected_packages
+            )
+        except KeyError:
+            return iter([])
```

Comment:
Is there a possibility that we have skipped data silently? Should this be logged? when can this happen?